queries/elixir.scm

```diff
@@ -0,0 +1,121 @@
+(comment) @comment
+(anonymous_function) @anonymousFunction
+(call
+  target: (_) @functionCallee
+) @functionCall @functionCallee.domain
+
+;; Map
+;; %{:a => "lorem", "b" => "ipsum", 3 => "dolor"}
+(map_content
+  (_)? @_.leading.start.endOf
+  .
+  (binary_operator) @collectionItem @_.leading.end.startOf @_.trailing.start.endOf
+  .
+  (_)? @_.trailing.end.startOf
+  (#insertion-delimiter! @collectionItem ", ")
+)
+(map_content
+  (binary_operator
+    left: (_) @collectionKey
+    right: (_) @value
+  ) @collectionKey.domain @value.domain
+)
+
+;; Shorthand map syntax
+;; %{a: "lorem", b: "ipsum"}
+(map_content
+  (keywords
+    (_)? @_.leading.start.endOf
+    .
+    (pair) @collectionItem @_.leading.end.startOf @_.trailing.start.endOf
+    .
+    (_)? @_.trailing.end.startOf
+  )
+  (#insertion-delimiter! @collectionItem ", ")
+)
+(map_content
+  (keywords
+    (pair
+      key: (_) @collectionKey
+      value: (_) @value
+    ) @collectionKey.domain @value.domain
+  )
+)
+
+(call
+  target: (identifier) @_target
+  (#match? @_target "^(def|defp|defdelegate|defmacro|defmacrop|defn|defnp)$")
+  (arguments
+    [
+      ; zero-arity functions with no parentheses
+      (identifier) @functionName
+      ; regular function clause
+      (call
+        target: (identifier) @functionName
+      )
+      ; function clause with a guard clause
+      (binary_operator
+        left: (call
+          target: (identifier) @functionName
+        )
+        operator: "when"
+      )
+    ]
+  )
+  (do_block) @namedFunction.interior
+  (#shrink-to-match! @namedFunction.interior "^do\\n?\\w*(?<keep>.*?\\n)\\s*end$")
+) @namedFunction @functionName.domain
+
+;; def fun(), do: 1
+(call
+  target: (identifier) @_target
+  (#match? @_target "^(def|defp|defdelegate|defmacro|defmacrop|defn|defnp)$")
+  (arguments
+    (keywords
+      (pair
+        key: (_) @do_keyword
+        value: (_) @namedFunction.interior
+      )
+    )
+    (#match? @do_keyword "^do: $")
+  )
+) @namedFunction @functionName.domain
+
+;; defguard guard(term) when is_integer(term) and rem(term, 2) == 0
+(call
+  target: (identifier) @_target
+  (#match? @_target "^(defguard|defguardp)$")
+  (arguments
+    (binary_operator
+      left: (call
+        target: (identifier) @functionName
+      )
+      operator: "when"
+    )
+  )
+) @namedFunction @functionName.domain
+
+(call
+  target: (identifier) @_target
+  (#match? @_target "^(defmodule)$")
+  (arguments
+    (alias) @className
+  )
+  (do_block
+    .
+    "do" @class.interior.start.endOf
+    "end" @class.interior.end.startOf
+    .
+  )
+) @class @className.domain
+
+(source) @className.iteration @class.iteration
+(source) @statement.iteration
+
+(source) @namedFunction.iteration @functionName.iteration
+;; Is it better to have the class as the function iteration scope?
```

Comment:
you want both the file and the class to be function iteration scope

Reply:
I believe I've done it, I don't use iteration stuff myself yet and I don't even know how to test if I've done it right 😅